tests/Downloader/DownloaderTest.php

```diff
@@ -333,4 +333,61 @@ public function testDontCallParseCallbackIfProcessedResponseWasDroppedByExtensio
 
         self::assertFalse($called);
     }
+
+    public function testDontSendRequestIfHasResponse(): void
+    {
+        $request = $this->makeRequest();
+        $request = $request->withResponse(
+            new Response(
+                new \GuzzleHttp\Psr7\Response(200, [], 'test'),
+                $request,
+            ),
+        );
```

Comment:
Can be shortened to

```suggestion
        $request = $request->withResponse($this->makeResponse($request));
```

This applies to the other tests as well.